### PR TITLE
IMPRO-1523: Convert optical flow to clize

### PR DESCRIPTION
**Pull request by neilCrosswaite**
As this is part of nowcasting it will not be going into the clize feature branch.

## Major change.
- the three radar files now get input as separate cubes and merged into a cubelist.
This is becase clize saw them as variable length and it clased with the orographic enhancement cube which could be `None`.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/tests/acceptance/test_nowcast_optical_flow.py  -1
         

Clones removed
==============
+ lib/improver/tests/acceptance/test_nowcast_optical_flow.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

**Comment by cgsandford** [comment]
@neilCrosswaite Following our brief chat, I need to be clearer on what this CLI should be doing:

- The radar input will come as multiple files, each containing one validity time.  These should therefore be read in a one-cube-per-file format using `load_cubelist` (via clize wrapping), which will not require any slicing.
- The orographic enhancement input will be a single file containing multiple lead times.  This also should not require any slicing as in theory this is handled by the plugin.

So this needs to be reworked please.  If there is any problem with the CLI test inputs, please discuss with me or @MoseleyS .

**Comment by neilCrosswaite** [comment]
> @neilCrosswaite Following our brief chat, I need to be clearer on what this CLI should be doing:
> 
> * The radar input will come as multiple files, each containing one validity time.  These should therefore be read in a one-cube-per-file format using `load_cubelist` (via clize wrapping).  This does not require any slicing or merging of cubes in the CLI.
> * The orographic enhancement input will be a single file containing multiple lead times.  This also should not require any slicing as this is handled by the `ApplyOrographicEnhancement` plugin.
> 
> So this needs to be reworked please. If there is any problem with the CLI test inputs, please discuss with me or @MoseleyS .

@cgsandford Should the orographi enhancement file still be optional?

**Comment by cgsandford** [comment]
@neilCrosswaite Stephen and I have discussed this and decided that for this CLI the orographic enhancement field should be compulsory.  I'll be away from tomorrow so have asked @MoseleyS to take over my review and make sure this is OK so it can be merged before Christmas.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

**Comment by neilCrosswaite** [comment]
The remasked tests needs an orographic enhancement cube, the one we used for the other tests is the wrong time.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/tests/acceptance/test_nowcast_optical_flow.py  -1
+ lib/improver/cli/nowcast_optical_flow.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/tests/acceptance/test_nowcast_optical_flow.py  -1
+ lib/improver/cli/nowcast_optical_flow.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

File: lib/improver/cli/nowcast_optical_flow.py [hunk 1]
```diff
@@ -32,96 +32,33 @@
...
     Args:
-        original_cube_list (iris.cube.CubeList):
-            Cubelist from which to calculate optical flow velocities.
-            The cubes require a 'time' coordinate on which they are sorted,
-            so the order of cubes does not matter.
         orographic_enhancement_cube (iris.cube.Cube):
             Cube containing the orographic enhancement fields.
-            Default is None.
+        radar (iris.cube.CubeList):
+            Cube from which to calculate optical flow velocities.
```
Comment (cgsandford) [inline lib/improver/cli/nowcast_optical_flow.py]: `Cubes`

Comment (neilCrosswaite) [reply lib/improver/cli/nowcast_optical_flow.py]:
```suggestion
            Cubes from which to calculate optical flow velocities.
```

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/tests/acceptance/test_nowcast_optical_flow.py  -1
+ lib/improver/cli/nowcast_optical_flow.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/tests/acceptance/test_nowcast_optical_flow.py  -1
+ lib/improver/cli/nowcast_optical_flow.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity decreasing per file
==============================
+ lib/improver/tests/acceptance/test_nowcast_optical_flow.py  -1
+ lib/improver/cli/nowcast_optical_flow.py  -4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4681969&bid=15659858)

**Pull request merged**